### ISSUE TITLE
IceCloneLocationAlreadyExists when destination folder exists but is empty

### ISSUE THREAD

**Issue by demarey** [npasserini/iceberg#1592]
`IceGitClone>>#execute` now starts with this check:
```smalltalk
location exists
		ifTrue: [ IceCloneLocationAlreadyExists signalFor: location ].
```

I propose to change it to:
```smalltalk
(location exists and: [ location hasChildren ])
		ifTrue: [ IceCloneLocationAlreadyExists signalFor: location ].
```
Indeed, why preventing to checkout in an empty directory?

**Comment by guillep** [comment]
Maybe, not a strong opinion on either :)

What I'm sure is that this check seems to fail a lot when a clone fails for the first time.
When a clone is attemted, the directory is created eagerly, and if this clone fails the directory does not get removed.
I dunno if the same issue or not, but surely related.

**Comment by demarey** [comment]
yes, I had the problem with a clone fail on git credentials missing.
The pb is that CI gives an IceCloneLocationAlreadyExists instead of an IceAuthenticationError that is more clear.

Anyway, cloning in an empty repository should be fine I think
